src/tss2-fapi/ifapi_keystore.c

```diff
@@ -627,6 +627,58 @@ ifapi_keystore_store_async(
     return r;
 }
 
+/**  Check whether a object path alread exists in keystore.
+ *
+ *  The FAPI path will be expanded to absolute path appropriate for
+ *  object and checked..
+ *
+ * @param[in] keystore The key directories and default profile.
+ * @param[in] path The relative path of the object. For keys the path will
+ *           expanded if possible.
+ * @param[in] object The object to be checked.
+ * @retval TSS2_RC_SUCCESS if the object does not exist.
+ * @retval TSS2_FAPI_RC_PATH_ALREADY_EXISTS: if the object exists.
+ * @retval TSS2_FAPI_RC_MEMORY: if memory could not be allocated to hold the output data.
+ * @retval TSS2_FAPI_RC_BAD_VALUE if an invalid value was passed into
+ *         the function.
+ * @retval TSS2_FAPI_RC_BAD_REFERENCE a invalid null pointer is passed.
+ */
+TSS2_RC
+ifapi_keystore_object_does_not_exist(
```

Comment:
The function name doesn't reflect the behavior or match the comments.
The function checks if object exists already, and so say the comments.
Should the name of the function be ifapi_keystore_object_exists()?

Reply:
The function returns TSS2_RC_SUCCESS if the object does not exist in keystore. So I would say the behaviour is OK.  So it was possible to use the standard mechanism for error checking instead of "if (ifapi_keystore_object_exists(..) == TSS2_RC_SUCCESS) error_handling."
Would it be sufficient if the comment is changed and describes exactly this behaviour or would you prefer the example above?

Reply:
It would be good to make that clear in the comment.